macros/core/src/shared.rs

```diff
@@ -175,3 +176,17 @@ pub(crate) fn generate_unexpected_input_panic(input_ident: &Ident, message: &str
         panic!(#message_pattern, input)
     })
 }
+
+pub(crate) fn extract_lifetime_names(type_args: AngleBracketedGenericArguments) -> Vec<String> {
```

Comment:
I would pass `&PathArguments`
```suggestion
pub(crate) fn extract_lifetime_names(type_args: &PathArguments) -> Vec<String> {
```